### PR TITLE
Reorder internal convolution and pooling operators for readability

### PR DESCRIPTION
**Pull request by jon-tow**
Convolution and pooling functions were scattered around the NN.swift file along with their corresponding vjps. This PR gathers them in an organized way for good code hygiene. The only non-reordering change was a line break deletion.

### CODE REVIEW

**Comment by jon-tow** [comment]
Glad to help!

File: Sources/TensorFlow/Operators/NN.swift [hunk 1]
```diff
@@ -316,24 +411,6 @@ func depthwiseConv2dBackpropInput<Scalar: TensorFlowFloatingPoint>(
...
-    return Raw.depthwiseConv2dNativeBackpropFilter(
-        x,
-        filterSizes: filterSizes,
-        outBackprop: x,
-        strides: [Int32(strides.0), Int32(strides.1), Int32(strides.2), Int32(strides.3)],
-        padding: padding.raw)
-}
-
 @usableFromInline
 func _vjpdepthwiseConv2dBackpropInput<Scalar: TensorFlowFloatingPoint>(
```
Comment (eaplatanios) [inline Sources/TensorFlow/Operators/NN.swift]:
```suggestion
func _vjpDepthwiseConv2dBackpropInput<Scalar: TensorFlowFloatingPoint>(
```

File: Sources/TensorFlow/Operators/NN.swift [hunk 2]
```diff
@@ -353,6 +430,24 @@ func _vjpdepthwiseConv2dBackpropInput<Scalar: TensorFlowFloatingPoint>(
     })
 }
 
+/// TensorFlow builtin depthwiseConv2D gradient helper for the filter.
+@differentiable(wrt: (x, input), vjp: _vjpdepthwiseConv2dBackpropFilter)
```
Comment (eaplatanios) [inline Sources/TensorFlow/Operators/NN.swift]:
```suggestion
@differentiable(wrt: (x, input), vjp: _vjpDepthwiseConv2dBackpropFilter)
```

File: Sources/TensorFlow/Operators/NN.swift [hunk 3]
```diff
@@ -353,6 +430,24 @@ func _vjpdepthwiseConv2dBackpropInput<Scalar: TensorFlowFloatingPoint>(
...
+    return Raw.depthwiseConv2dNativeBackpropFilter(
+        x,
+        filterSizes: filterSizes,
+        outBackprop: x,
+        strides: [Int32(strides.0), Int32(strides.1), Int32(strides.2), Int32(strides.3)],
+        padding: padding.raw)
+}
+
 @usableFromInline
 func _vjpdepthwiseConv2dBackpropFilter<Scalar: TensorFlowFloatingPoint>(
```
Comment (eaplatanios) [inline Sources/TensorFlow/Operators/NN.swift]:
```suggestion
func _vjpDepthwiseConv2dBackpropFilter<Scalar: TensorFlowFloatingPoint>(
```

**Comment by eaplatanios** [comment]
Note that the renaming changes I suggested will also require you to change the references to these VJP functions from other places.

File: Sources/TensorFlow/Operators/NN.swift (hunk 2 above)
Comment (jon-tow) [reply Sources/TensorFlow/Operators/NN.swift]: Hey @eaplatanios. Thanks for helping me out. I'm not sure what to change here? Could you please clarifiy.

Comment (jon-tow) [reply Sources/TensorFlow/Operators/NN.swift]: @eaplatanios Good eye! I didn't realize that wasn't camel case. I wasn't paying too much attention to naming when moving things around. Thanks!

**Comment by rxwei** [comment]
> 😕 The bad news is that it appears that one or more commits were authored or co-authored by someone other than the pull request submitter. We need to confirm that all authors are ok with their commits being contributed to this project. Please have them confirm that here in the pull request.

CLA doesn't work automatically with directly-committed suggestions. For formality, @eaplatanios could you please confirm that you are okay with your commits being contributed to this PR?

**Comment by eaplatanios** [comment]
Yes sure. I confirm that I am okay with my commits being contributed to this PR.

**Comment by jon-tow** [comment]
Sorry I'm trying to remove the merge branch commits to clean up the commit history.

**Comment by jon-tow** [comment]
@rxwei I had to remove some "merge branch" commits that cluttered the PR's commit history just in case it bothers you all as much as it bothers me :)

**Pull request merged**